### ISSUE TITLE
Loss function does not decrease and weights are not ben updated

### ISSUE THREAD

**Issue by victorc365** [tensorflow/recommenders#231]
I tried the retrieval model as in the tutorial with another dataset and loss function does not decrease and weights are not updated with each epoch, also after training with several epochs I found that all embeddings are identical. Someone can help me to know why is this happening? How can I fix it? 
I'm training only with ids of users and items as in tutorial.

**Comment by turnerneil** [comment]
It could be due to your dataset or hyperparameters. I was seeing a similar issue when I had a highly sparse dataset with few interactions per user.

This ticket is similar: https://github.com/tensorflow/recommenders/issues/134

**Comment by maciejkula** [comment]
As @turnerneil said, it's likely your dataset of hyparparameters. A lot of things can go wrong in deep learning, it's the nature of the beast.
